modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PlacementDriver.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.table.distributed.replicator;
+
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import org.apache.ignite.internal.replicator.ReplicaService;
+import org.apache.ignite.internal.tx.TxMeta;
+import org.apache.ignite.internal.tx.message.TxStateReplicaRequest;
+import org.apache.ignite.lang.IgniteBiTuple;
+import org.apache.ignite.network.ClusterNode;
+
+/**
+ * Placement driver.
+ */
+public class PlacementDriver {
+    /** Assignment nodes per replication group. */
+    private final Map<String, LinkedHashSet<ClusterNode>> primaryReplicaMapping = new ConcurrentHashMap<>();
+
+    /** Replication service. */
+    private final ReplicaService replicaService;
+
+    /**
+     * The constructor.
+     *
+     * @param replicaService Replication service.
+     */
+    public PlacementDriver(ReplicaService replicaService) {
+        this.replicaService = replicaService;
+    }
+
+    /**
+     * Sent a transaction sate request to the primary replica.
+     *
+     * @param replicaGrp Replication group id.
+     * @param request Status request.
+     * @return Result future.
+     */
+    public CompletableFuture<TxMeta> sentMetaRequest(String replicaGrp, TxStateReplicaRequest request) {
+        CompletableFuture<TxMeta> resFut = new CompletableFuture<>();
+
+        sentAndRetry(resFut, replicaGrp, request);
+
+        return resFut;
+    }
+
+    /**
+     * Updates an assignment for the specific replication group.
+     *
+     * @param replicaGrpId Replication group id.
+     * @param assignment Assignment.
+     */
+    public void updateAssignment(String replicaGrpId, Collection<ClusterNode> assignment) {
+        primaryReplicaMapping.put(replicaGrpId, new LinkedHashSet<>(assignment));
+    }
+
+    /**
+     * Sent a request to primary replica of the replication group.
```

Comment:
the first replica, not a primary.

Reply:
It tries to send it on primary node.
I tried to explain those concept in another terms, please look it again.

Reply:
That's better